--- plotseditor.cpp	2013-03-10 01:11:28.000000000 -0500
+++ plotseditor_changed.cpp	2013-03-10 01:00:58.000000000 -0500
@@ -705,10 +705,10 @@
                 } else
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
                 
-                item->setInterval(item->parameters().first(), m_widget->minx->expression(), m_widget->maxx->expression());
+              //  item->setInterval(item->parameters().first(), m_widget->minx->expression(), m_widget->maxx->expression());
 
                 if(!isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else
                 errors = req.errors();
 
@@ -726,11 +726,11 @@
                     item = editCurrentFunction(req.expression());
                 } else
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
-                item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
-                item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
+              //  item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
+              //  item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
 
                 if (isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else {
                 errors = req.errors();
 			}
@@ -748,11 +748,11 @@
                 } else {
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
                 }
-                item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
-                item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
+               // item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
+               // item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
 
                 if (!isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else
                 errors = req.errors();
 
@@ -769,12 +769,12 @@
                 } else {
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
                 }
-                item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
-                item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
-                item->setInterval(item->parameters().at(2), m_widget->minz->expression(), m_widget->maxz->expression());
+           //     item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
+            //    item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
+             //   item->setInterval(item->parameters().at(2), m_widget->minz->expression(), m_widget->maxz->expression());
 
                 if (!isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else
                 errors = req.errors();
 
@@ -792,10 +792,10 @@
                 } else {
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
                 }
-                item->setInterval(item->parameters().first(), m_widget->minx->expression(), m_widget->maxx->expression());
+           //     item->setInterval(item->parameters().first(), m_widget->minx->expression(), m_widget->maxx->expression());
 
                 if (!isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else
                 errors = req.errors();
 
@@ -812,10 +812,10 @@
                 } else {
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
                 }
-                item->setInterval(item->parameters().first(), m_widget->minx->expression(), m_widget->maxx->expression());
+           //     item->setInterval(item->parameters().first(), m_widget->minx->expression(), m_widget->maxx->expression());
 
                 if (!isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else
                 errors = req.errors();
             
@@ -832,11 +832,11 @@
                 } else {
                     item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
                 }
-                item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
-                item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
+             //   item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
+              //  item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
 
                 if (!isEditing)
-                    m_document->plotsModel()->addPlot(item);
+                    m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
                 
             } else
                 errors = req.errors();
