--- plotseditor.cpp	2013-03-25 01:50:26.031701717 -0500
+++ plotseditor_new.cpp	2013-03-25 01:52:23.355697072 -0500
@@ -724,16 +724,16 @@
                 FunctionGraph *item = 0;
                 if (isEditing) {
                     item = editCurrentFunction(req.expression());
-                } else
-                    item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
-              //  item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
+                } else{
+                  item = req.create(m_widget->plotColor->color(), m_widget->plotName->text());
+             } //  item->setInterval(item->parameters().at(0), m_widget->minx->expression(), m_widget->maxx->expression());
               //  item->setInterval(item->parameters().at(1), m_widget->miny->expression(), m_widget->maxy->expression());
 
-                if (isEditing)
+                if (!isEditing)
                     m_document->plotsModel()->addPlot(req.create(m_widget->plotColor->color(), m_widget->plotName->text()));
             } else {
                 errors = req.errors();
-			}
+        }
 
             break;
         }
